File: src/api/endpoints/emergency/response/TeamDetailsResponse.ts
import { Level } from "../../../../models/Level";

/** Details about the responders assigned to an alert. */
export default interface TeamDetailsResponse {
  /** Statistics for each responder. */
  stats: ResponderDetails[];

  /** The success rate across all responders. */
  aggregatedSuccessRate: number;
}

/** Details about an alert responder. */
export interface ResponderDetails {
  /** The responder's id. */
  id: string;

  /** The responder's level. */
  level: Level;

  /** The success rate for missions to which this staff member responded. */
  missionSuccessRate: number;

  /** The success rate for missions this staff member dispatched. */
  dispatchSuccessRate: number;
}

File: src-dotnet/api/endpoints/emergency/response/TeamDetailsResponse.cs
using System.Collections.Generic;
using MedrunnerApiClient.Models;

namespace MedrunnerApiClient.Api.Endpoints.Emergency.Response;

    /// <summary>
    /// Details about a team responding to an alert.
    /// </summary>
    public class TeamDetailsResponse
    {
        /// <summary>
        /// Details about each individual responder.
        /// </summary>
        public List<ResponderDetails> Stats { get; set; }

        /// <summary>
        /// The aggregated mission success rate from all responders, appropriately weighted by number of missions.
        /// </summary>
        public double AggregatedSuccessRate { get; set; }
    }

    /// <summary>
    /// Details about an alert responder.
    /// </summary>
    public class ResponderDetails
    {
        /// <summary>
        /// The responder's id.
        /// </summary>
        public string Id { get; set; }

        /// <summary>
        /// The responder's level.
        /// </summary>
        public Level Level { get; set; }

        /// <summary>
        /// The success rate of all prior missions this staff member has responded to.
        /// </summary>
        public double MissionSuccessRate { get; set; }

        /// <summary>
        /// The success rate of all prior missions this staff member has acted as a dispatcher for.
        /// </summary>
        public double DispatchSuccessRate { get; set; }
    }
    